style.js

```diff
@@ -0,0 +1,659 @@
+        "body": "eum non blanditiis soluta porro quibusdam voluptas\nvel voluptatem qui placeat dolores qui velit aut\nvel inventore aut cumque culpa explicabo aliquid at\nperspiciatis est et voluptatem dignissimos dolor itaque sit nam"
+    },
+    {
+        "userId": 10,
+        "id": 98,
+        "title": "laboriosam dolor voluptates",
+        "body": "doloremque ex facilis sit sint culpa\nsoluta assumenda eligendi non ut eius\nsequi ducimus vel quasi\nveritatis est dolores"
+    },
+    {
+        "userId": 10,
+        "id": 99,
+        "title": "temporibus sit alias delectus eligendi possimus magni",
+        "body": "quo deleniti praesentium dicta non quod\naut est molestias\nmolestias et officia quis nihil\nitaque dolorem quia"
+    },
+    {
+        "userId": 10,
+        "id": 100,
+        "title": "at nam consequatur ea labore ea harum",
+        "body": "cupiditate quo est a modi nesciunt soluta\nipsa voluptas error itaque dicta in\nautem qui minus magnam et distinctio eum\naccusamus ratione error aut"
+    }
+];
+
+function populateArticles(articleList) {
+    let articlesSection = document.querySelector(".articles-area");
+    articlesSection.textContent = "";
+
+    //Run a loop through the products and create an HTML element
+    articles.forEach(function (article, index) {
+        //Create the HTML element for the individual product
+        let articleElm = document.createElement("div");
+        articleElm.classList.add("article-item");
+
+        //Create the product details section
+        let articleDetails = document.createElement("div");
+        articleDetails.classList.add("article-details");
+
+        //Create id article
+        let articleId = document.createElement("p");
+        articleId.classList.add("article-id");
+        articleId.textContent = article.id;
+        let articleId2 = localStorage.setItem('id', article.id);
+
+        //Create the article title
+        let articleTitle = document.createElement("h1");
+        articleTitle.classList.add("article-title");
+        articleTitle.textContent = article.title;
+
+        //Create article paragraph
+        let articleParagraph = document.createElement("p");
+        articleParagraph.classList.add("article-paragraph");
+        articleParagraph.textContent = article.body;
+
+        articleDetails.append(articleId);
+        articleDetails.append(articleTitle);
+        articleDetails.append(articleParagraph);
+
+        articleElm.append(articleDetails);
+
+        articlesSection.append(articleElm);
+    })
+}
+
+function articlesHandler() {
+    populateArticles(articles);
+
+    document.querySelector(".article-title").addEventListener("click", function (e) {
+        /* La mia soluzione era di prendere il link 
+        della pagina esterna e inserirci all'interno dell'url l'id dell'articolo attuale però ogni volta che faccio il localStorage.getItem("id") per prendermi l'id mi da 100 */
```

Comment:
The idea is correct, the error is the place where you set the id in the localStorage. Doing it in the for the id is overwritten as the array is cycled. By setting the id in the localStorage on click on the item you would have the correct id.